lib/Reactify.re

```diff
@@ -382,7 +379,7 @@ module Make = (ReconcilerImpl: Reconciler) => {
             /* Only both replacing node if the primitives are different */
             switch (newInstance.component.element, i.component.element) {
             | (Primitive(newPrim), Primitive(oldPrim)) =>
-              if (oldPrim != newPrim) {
+              if (oldPrim !== newPrim) {
```

Comment:
This is probably the most relevant change. Because we were using structural equality before, when the primitives would contain any function in side (like, say, a `Button` that has a callback `onPress`) an exception with `Invalid_argument equal: functional value` would be thrown.

I am not sure about the perf implications. This change will lead to the reconciler rendering much more often but on the other hand the checks should be much faster too.

Reply:
Ah interesting, thanks for calling it out! Would've been easy to miss 😄 

>  Because we were using structural equality before, when the primitives would contain any function in side (like, say, a Button that has a callback onPress) an exception with Invalid_argument equal: functional value would be thrown.

Good catch! Yes, that makes sense that it is problematic in this case. Comparing 'functions' is not really doable in the OCaml sense like it is in JS.

> I am not sure about the perf implications. This change will lead to the reconciler rendering much more often but on the other hand the checks should be much faster too.

Yes, good point, I suspect it will depend on how much work it is to create a new 'node' too. Would really need benchmarks to decide if it is a bottleneck or not.

But I'm OK with this going in since it behaves correctly and all the tests are green. We can focus on addressing the performance once we find bottlenecks / have benchmarks to examine. I created #37 to examine this in more depth.